### ISSUE TITLE
Report unchecked type-assertion when returning value and when passing it to other func

### ISSUE THREAD

**Issue by leonklingele** [kisielk/errcheck#224]
Two false negatives here when using the `-asserts` flag:

```go
package main

func main() {
	var i interface{} = "hello"

	_ = handleInterface(i)

	handleString(i.(string)) // This should be reported but isn't
}

func handleInterface(i interface{}) string {
	return i.(string) // This should be reported but isn't
}

func handleString(s string) {}
```

**Comment by leonklingele** [comment]
Also, this is not caught:

```go
if i.(string) == "hello" {
    ..
}
```
